test/test_connectionpool.py

```diff
@@ -283,6 +283,28 @@ def test_ca_certs_default_cert_required(self):
             conn = pool._get_conn()
             self.assertEqual(conn.cert_reqs, 'CERT_REQUIRED')
 
+    def test_cleanup_on_extreme_connection_error(self):
+        class RealBad(BaseException):
```

Comment:
So, perhaps it's just me, but I'd love to have a comment or docstring for this that describes why we have an exception class inheriting from BaseException and why we're testing that here.

(In reality, I'd love for all tests to have docstrings, but this one especially.)